src/components/PostSelector/PostSelector.js

```diff
@@ -0,0 +1,236 @@
+  const [suggestions, setSuggestions] = useState([])
+  const [offset, setOffset] = useState('')
+  const [selectedPosts, setSelectedPosts] = useState(posts || [])
+  const [suggestionsOpen, setSuggestionsOpen] = useState(false)
+  const searchBox = useRef(false)
+
+  const debouncedAutcomplete = useDebounce(autocomplete, 300)
+
+  useEffect(() => setSelectedPosts(posts || []), [posts])
+
+  useEffect(() => {
+    // TODO: tell people they need to type 2 characters to get results?
+    if (!autocomplete?.length || autocomplete.length > 1) {
+      dispatch(fetchPosts({ collectionToFilterOut: collection?.id, context: 'groups', slug: group.slug, search: debouncedAutcomplete, sortBy: 'created', first: PAGE_SIZE, offset: 0 })).then(res => {
+        setSuggestions(res.payload?.data?.group?.posts?.items || [])
+      })
+      setOffset(PAGE_SIZE)
+    }
+  }, [debouncedAutcomplete, collection])
+
+  const { observe } = useInView({
+    onEnter: () => {
+      dispatch(fetchPosts({ collectionToFilterOut: collection?.id, context: 'groups', slug: group.slug, search: debouncedAutcomplete, sortBy: 'created', first: PAGE_SIZE, offset: offset })).then(res => {
+        setSuggestions(suggestions.concat(res.payload?.data?.group?.posts?.items || []))
+        setOffset(offset + PAGE_SIZE)
+      })
+    }
+  })
+
+  const hideSuggestions = () => setSuggestionsOpen(false)
+
+  useEffect(() => {
+    if (suggestionsOpen) {
+      // initiate the event handler
+      window.addEventListener('click', hideSuggestions)
+
+      // this will clean up the event every time the component is re-rendered
+      return function cleanup () {
+        window.removeEventListener('click', hideSuggestions)
+      }
+    }
+  }, [suggestionsOpen])
+
+  const handleInputChange = (input) => {
+    setAutocomplete(input)
+  }
+
+  const handleDelete = (post, index) => () => {
+    if (window.confirm('Remove post?')) {
+      setSelectedPosts((prevPosts) => {
+        onRemovePost(post)
+        return update(prevPosts, {
+          $splice: [
+            [index, 1]
+          ]
+        })
+      })
+    }
+  }
+
+  const handleSelectPost = (p, event) => {
+    if (onSelectPost) {
+      onSelectPost(p)
+    }
+    setSelectedPosts([...selectedPosts].concat(p))
+    searchBox.current.focus()
+    event.preventDefault()
+    event.stopPropagation()
+  }
+
+  const movePost = useCallback((dragIndex, hoverIndex) => {
+    setSelectedPosts((prevPosts) => {
+      onReorderPost(prevPosts[dragIndex], hoverIndex)
+      return update(prevPosts, {
+        $splice: [
+          [dragIndex, 1],
+          [hoverIndex, 0, prevPosts[dragIndex]]
+        ]
+      })
+    })
+  }, [])
+
+  const displaySuggestions = useMemo(() => {
+    const selectedPostIds = selectedPosts.map(p => p.id)
+    return suggestions.filter(s => !selectedPostIds.includes(s.id))
+  }, [suggestions, selectedPosts])
+
+  const pending = useSelector(state => isPendingFor(FETCH_POSTS, state))
```

Comment:
In functional components I tend to rely upon a local `loading` state variable and ignore the `PENDING` middleware stuff. I think I do that as I can be more granular about things and also because I feel like I'm always trying to wean away from depending on Redux'y stuff when the alternative is as simple or simpler. If that makes sense. Don't do this now. This is fine. Just taking the chance to have some code chats async...